core/src/client/async_client/mod.rs

```diff
@@ -21,9 +20,11 @@ use async_lock::Mutex;
 use async_trait::async_trait;
 use futures_channel::{mpsc, oneshot};
 use futures_timer::Delay;
-use futures_util::future::{self, Either};
+use futures_util::future::{self, Either, Fuse};
+use futures_util::select;
```

Comment:
you have to add the `async-await-macro` feature on `future_util` in the `async-wasm-client` for that to work.

Reply:
It seems that `select` depends on the `std` feature, which I believe we don't want as default (https://github.com/rust-lang/futures-rs/blob/master/futures-util/src/async_await/select_mod.rs#L306).

Have switched to `select_biased` which would go through the futures in order we have them defined (instead of picking a random path)